src/lib/arch/aarch64/Instruction.cc

```diff
@@ -36,6 +36,11 @@ InstructionException Instruction::getException() const { return exception_; }
 const span<Register> Instruction::getOperandRegisters() const {
   return {const_cast<Register*>(sourceRegisters.data()), sourceRegisterCount};
 }
+
+const span<RegisterValue> Instruction::getSourceOperands() const {
+  return {const_cast<RegisterValue*>(operands.data()), operands.size()};
+}
+
```

Comment:
Any specific reason to be returning a span rather than a vector as in the AArch64/Instruction.hh implementation of operands, possibly preventing this awkward conversion. The RISC-V implementation has operands as an array so conversion would be necessary either way

Reply:
Using span as a return type was copied from existing code in the class (see `getOperandRegisters()`). 

In terms of why a span - if we return a vector then we create a new copy of the data, whilst a span (made up of a pointer and length) will just point to said data.